Client/mods/deathmatch/logic/luadefs/CLuaGifDefs.cpp

```diff
@@ -0,0 +1,359 @@
+        CResource* resource = luaMain.GetResource();
+        SString path;
+        SString metaPath;
+        SString fileBuffer;
+        SString lastPath = buffer;
+        SString format = buffer.substr(0,6);
+        CClientGifLoader loader;
+        if (format == "GIF87a" || format == "GIF89a") {
+            loader = CClientGifLoader(buffer);
+        }else if (CResourceManager::ParseResourcePathInput(buffer,resource,&path,&metaPath)){
+            if (FileExists(path)) {
+                eAccessType  access = buffer[0] == '@' ? eAccessType::ACCESS_PRIVATE : eAccessType::ACCESS_PUBLIC;
+                CScriptFile* file = new CScriptFile(resource->GetScriptID(), metaPath, DEFAULT_MAX_FILESIZE,access);
+                if (file->Load(resource, CScriptFile::MODE_READ)) {
+                    file->SetLuaDebugInfo(g_pClientGame->GetScriptDebugging()->GetLuaDebugInfo(luaVM));
+                    long size = file->GetSize();
+                    long bytesRead = file->Read(size,fileBuffer);
+                    if (bytesRead >= 0){
+                        loader = CClientGifLoader(fileBuffer);
+                    }else if (bytesRead == -2){ 
+                        m_pScriptDebugging->LogWarning(luaVM, "out of memory");
+                    }else{
+                        m_pScriptDebugging->LogError(luaVM, SString("error while reading file [%s]", lastPath.c_str()).c_str());
+                    }
+                    file->Unload();
+                    m_pElementDeleter->Delete(file);
+                }else{
+                    delete file;
+                    m_pScriptDebugging->LogError(luaVM, SString("couldn't load file [%s]",lastPath.c_str()).c_str());
+                    lua_pushboolean(luaVM, false);
+                    return 1;
+                }
+            }else{
+                m_pScriptDebugging->LogError(luaVM, SString("file [%s] doesn't exists!",lastPath.c_str()).c_str());
+                lua_pushboolean(luaVM, false);
+                return 1;
+            }
+        }else{
+            loader = CClientGifLoader(buffer);
+        }
+        std::vector<unsigned char*> frames;
+        std::vector<int> delays;
+        loader.Load(frames, delays, 0L);
+        if (!loader) {
+            m_pScriptDebugging->LogError(luaVM, "wrong file format");
+            lua_pushboolean(luaVM, false);
+            return 1;
+        }
+        if (frames.size() < 1) {
+            m_pScriptDebugging->LogError(luaVM, "gif has no frames");
+            lua_pushboolean(luaVM,false);
+            return 1;
+        }
+        uint width = (uint)loader.GetWidth();
+        uint height = (uint)loader.GetHeight();
+        if (width < 1 || height < 1) {
+            m_pScriptDebugging->LogError(luaVM, "gif must be 1x1 at least");
+            lua_pushboolean(luaVM,false);
+            return 1;
+        }
+        CClientGif* gif = g_pClientGame->GetManager()->GetRenderElementManager()->CreateGif(width,height);
```

Comment:
Missing space here

Reply:
i couldn't understand what you mean from 'Missing space'

Reply:
The space between **width** and **height**

Reply:
aha fixed